packages/tools/viewer/src/viewer.ts

```diff
@@ -1411,9 +1608,127 @@ export class Viewer implements IDisposable {
+        const handleLoadError = async (promise: Promise<unknown>) => {
+            try {
+                await promise;
+            } catch (error) {
+                if (!(error instanceof AbortError)) {
+                    Logger.Error(error);
+                }
+            }
+        };
+
+        if (flags.length === 0 || flags.includes("source")) {
+            handleLoadError(this._updateModel(this._options?.source));
+        }
+
+        if (flags.length === 0 || flags.includes("environment")) {
+            this._scene.clearColor = new Color4(...(this._options?.clearColor ?? DefaultViewerOptions.clearColor));
+            this.environmentConfig = {
+                intensity: this._options?.environmentConfig?.intensity ?? DefaultViewerOptions.environmentConfig.intensity,
+                blur: this._options?.environmentConfig?.blur ?? DefaultViewerOptions.environmentConfig.blur,
+                rotation: this._options?.environmentConfig?.rotation ?? DefaultViewerOptions.environmentConfig.rotation,
+                visible: this._options?.environmentConfig?.visible ?? DefaultViewerOptions.environmentConfig.visible,
+            };
+            if (this._options?.environmentLighting === this._options?.environmentSkybox) {
+                handleLoadError(this._updateEnvironment(this._options?.environmentLighting, { lighting: true, skybox: true }, this._loadEnvironmentAbortController?.signal));
+            } else {
+                handleLoadError(this._updateEnvironment(this._options?.environmentLighting, { lighting: true }, this._loadEnvironmentAbortController?.signal));
+                handleLoadError(this._updateEnvironment(this._options?.environmentSkybox, { skybox: true }, this._loadSkyboxAbortController?.signal));
+            }
+        }
+
+        if (flags.length === 0 || flags.includes("animation")) {
+            this.animationSpeed = this._options?.animationSpeed ?? DefaultViewerOptions.animationSpeed;
+            this.selectedAnimation = this._options?.selectedAnimation ?? 0;
+            if (this._options?.animationAutoPlay) {
+                this.playAnimation();
+            } else {
+                this.pauseAnimation();
+            }
+        }
+
+        if (flags.length === 0 || flags.includes("camera")) {
+            // In the case of resetting the camera, we always want to restore default states, so convert NaN to undefined.
+            const alpha = Number(this._options?.cameraOrbit?.[0]);
+            const beta = Number(this._options?.cameraOrbit?.[1]);
+            const radius = Number(this._options?.cameraOrbit?.[2]);
+            const targetX = Number(this._options?.cameraTarget?.[0]);
+            const targetY = Number(this._options?.cameraTarget?.[1]);
+            const targetZ = Number(this._options?.cameraTarget?.[2]);
+            this._reframeCameraFromBounds(
+                interpolate,
+                this._loadedModels,
+                isNaN(alpha) ? undefined : alpha,

```

Comment:
See comment above (probably no need for the `isNaN` check?)

Reply:
Same thing. Let me know if I should clarify the comment just above this block of code.